_posts/2023-05-08-changelog-194.markdown

```diff
@@ -0,0 +1,132 @@
+---
+layout: post
+title: "IntelliJ Rust Changelog #194"
+date: 2023-05-08 13:00:00 +0300
+---
+
+
+## New Features
+
+* Detect new compiler errors:
+
+  * [#10406] The `not` cfg-predicate was malformed [E0536](https://doc.rust-lang.org/error_codes/E0536.html)
+
+  * [#10387] A non-ident or non-wildcard pattern has been used as a parameter of a function pointer type [E0561](https://doc.rust-lang.org/error_codes/E0561.html)
+
+  * [#10399] `async` non-`move` closures with parameters are currently not supported [E0708](https://doc.rust-lang.org/error_codes/E0708.html)
+
+  * [#10361] Variadic parameters have been used on a non-C ABI function [E0045](https://doc.rust-lang.org/error_codes/E0045.html)
+
+* [#10315] Support extracting code containing control flow in [`Extract function`](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-refactorings.html#extractmethod-refactoring) refactoring (`Refactor | Extract Method` or <kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>M</kbd>)
+
+  {% include gif-img.html path="/assets/posts/changelog-194/extract_function_control_flow" w="700px" %}
+
+* [#9219] Adjust new Rust project generator to new project wizard in IDEA
+
+  <img src="/assets/posts/changelog-194/new_project_wizard.png" width="700px"/>
+
+* [#10398] Highlight constants and statics differently to visually distinguish them
+
+  <img src="/assets/posts/changelog-194/static_const.png" width="350px"/>
+
+* [#10359] Annotate built-in attributes (by [@kuksag])
+
+  * Check that the proper delimiter is applied when specifying an attribute
+
+  * Check that attribute format is conforming with the respective template
+
+  * Check that no literal suffixes are used when specifying an attribute
+
+  * Check if an attribute is used multiple times on the same item
+
+* [#9551] Allow to configure the channel and environment variables for external linters
+
+* [#7509] Add support for multiple spans in external linter check inspection (by [@Kobzol])
```

Comment:
Should be removed since it is reverted by https://github.com/intellij-rust/intellij-rust/pull/10430